### ISSUE TITLE
Url Parameters are not checked in BuildNavigationPlanStep

### ISSUE THREAD

**Issue by janosch715** [aurelia/router#268]
If navigating from e.g. `/app.html#/page1` to `/app.html#/page1?param1=abc` the `no-change` view port plan is choosen in the `BuildNavigationPlanStep` function. So no view model is ever informed about the navigation.
The problem is that the function `hasDifferentParameterValues`, called from `BuildNavigationPlanStep`, compares the `params` of the current `NavigationInstruction` object to the previous `NavigationInstruction` object, but not the `queryParams`.
I think the `queryParams` should also be checked so that the `invoke-lifecycle` view port plan is choosen when navigating from `/app.html#/page1` to `/app.html#/page1?param1=abc`.
I am using `npm:aurelia-router@1.0.0-beta.1` and referring to `navigation-plan.js` of the latest commit (c6af5d1).

**Comment by EisenbergEffect** [comment]
@bryanrsmith Any thoughts on this? I know it can be handled by changing the strategy today, but should we have some sort of route config setting that tells the router to take query strings into account?

**Comment by bryanrsmith** [comment]
I agree with @janosch715. I think the current behavior is unexpected.

**Comment by jedd-ahyoung** [comment]
@EisenbergEffect @bryanrsmith A few thoughts - 

1. Any ideas for router configuration parameter names? Perhaps `config.ignoreQueryParameters = true` or something like that?
2. This is marked with the `breaking` tag. Will we be changing the query check to be the default behavior, or will this be an opt-in?

Other than these two issues, I should be able to knock this out fairly quickly.

**Comment by EisenbergEffect** [comment]
Can we keep today's behavior the default and then add a config setting to add the new behavior? Then, in a future version we could potentially change the default, after we go through a cycle of emitting warning of the planned change, etc. What do you think?

**Comment by jedd-ahyoung** [comment]
That sounds reasonable and I like the fact that can can provide an easy
migration plan.

2016-02-27 22:52 GMT-05:00 Rob Eisenberg <notifications@github.com>:

**Comment by janosch715** [comment]
Unfortunately the commit does not include the possibility to pass the options to the constructor of `Router`. 
I would also appreciate if the options can be passed directly during aurelia configuration like: 
```javascript
...
aurelia.use
        .basicConfiguration()
        .history()
        .router( { compareQueryParams: true } );
...
```
